script/hassfest/zeroconf.py

```diff
@@ -34,38 +34,10 @@ def generate_and_validate(integrations: Dict[str, Integration]):
         homekit = integration.manifest.get("homekit", {})
         homekit_models = homekit.get("models", [])
 
-        if not service_types and not homekit_models:
+        if not (service_types or homekit_models):
             continue
 
-        try:
-            with open(str(integration.path / "config_flow.py")) as fp:
-                content = fp.read()
-                uses_discovery_flow = "register_discovery_flow" in content
-                uses_oauth2_flow = "AbstractOAuth2FlowHandler" in content
-
-                if (
-                    service_types
-                    and not uses_discovery_flow
-                    and not uses_oauth2_flow
-                    and " async_step_zeroconf" not in content
-                ):
-                    integration.add_error(
-                        "zeroconf", "Config flow has no async_step_zeroconf"
-                    )
-                    continue
-
-                if (
-                    homekit_models
-                    and not uses_discovery_flow
-                    and not uses_oauth2_flow
-                    and " async_step_homekit" not in content
-                ):
-                    integration.add_error(
-                        "zeroconf", "Config flow has no async_step_homekit"
-                    )
-                    continue
-
-        except FileNotFoundError:
+        if not (integration.path / "config_flow.py").exists():
```

Comment:
I wonder if we should move this check to `hassfest/config_flow.py` and test against any of the `DISCOVERY_INTEGRATIONS` keys in manifest.

Reply:
Hmm... yeah, I think we can clean this up even more. I will look into this.

Reply:
Consolidated all hassfest checks related to configuration flow in the config_flow plugin.

The SSDP & Zeroconf plugins are now only for generating/validating their own generated sources.

I've opted not to use `DISCOVERY_INTEGRATIONS`, mainly because it is incomplete for this use case (e.g., missing hassio and generic discovery methods). I think it makes the code more readable + it found new cases now.